Windows Errors (py27 only)

In #698, I got more tests running on appveyor, including windows-32 and windows-64.

There are a couple errors.

Py27 / Win-32:

```
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_read_0
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_mdcrd.py", line 41, in test_read_0
    xyz, box = f.read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
 
ERROR: mdtraj.tests.test_mdcrd.test_read_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_mdcrd.py", line 51, in test_read_1
    xyz, _ = f.read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_multiread
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_mdcrd.py", line 90, in test_multiread
    reference = md.load(get_fn('frame0.mdcrd'), top=get_fn('native.pdb'))
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_seek
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_mdcrd.py", line 99, in test_seek
    reference = md.load(get_fn('frame0.mdcrd'), top=get_fn('native.pdb'))
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_atom_indices_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_mdcrd.py", line 132, in test_atom_indices_1
    t0 = md.load(get_fn('frame0.mdcrd'), top=top)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_trajectory.test_seek_read_mode
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_trajectory.py", line 398, in test_seek_read_mode
    xyz = md.load(get_fn(b), top=get_fn('native.pdb')).xyz
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_trajectory.test_load_frame
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_32\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\tests\test_trajectory.py", line 447, in test_load_frame
    trajectories = [md.load(get_fn(f), top=get_fn('native.pdb')) for f in files]
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_32\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win32.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
----------------------------------------------------------------------
Ran 1047 tests in 752.966s
 
FAILED (SKIP=24, errors=7)
```

Py27 / Win-64
```
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_read_0
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_mdcrd.py", line 41, in test_read_0
    xyz, box = f.read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_read_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_mdcrd.py", line 51, in test_read_1
    xyz, _ = f.read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_multiread
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_mdcrd.py", line 90, in test_multiread
    reference = md.load(get_fn('frame0.mdcrd'), top=get_fn('native.pdb'))
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_seek
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_mdcrd.py", line 99, in test_seek
    reference = md.load(get_fn('frame0.mdcrd'), top=get_fn('native.pdb'))
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_mdcrd.test_atom_indices_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_mdcrd.py", line 132, in test_atom_indices_1
    t0 = md.load(get_fn('frame0.mdcrd'), top=top)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_trajectory.test_seek_read_mode
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_trajectory.py", line 398, in test_seek_read_mode
    xyz = md.load(get_fn(b), top=get_fn('native.pdb')).xyz
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
ERROR: mdtraj.tests.test_trajectory.test_load_frame
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_trajectory.py", line 447, in test_load_frame
    trajectories = [md.load(get_fn(f), top=get_fn('native.pdb')) for f in files]
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\core\trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 100, in load_mdcrd
    xyz, cell_lengths = f.read(stride=stride, atom_indices=atom_indices)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 251, in read
    coord, box = self._read()
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\mdcrd.py", line 307, in _read
    'encountered. ' % self._n_atoms)
IOError: mdcrd parse error: specified n_atoms (22) is likely incorrect. Incorrect buffer size encountered. 
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_load_multiframe
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 69, in <lambda>
    yield lambda: eq(t.n_frames, 2)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_load_multiframe
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 70, in <lambda>
    yield lambda: eq(t.n_atoms, 22)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_4K6Q
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 76, in test_4K6Q
    eq(t.n_frames, 1)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_2EQQ_0
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 101, in <lambda>
    yield lambda: eq(t.n_frames, 20)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_2EQQ_0
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 102, in <lambda>
    yield lambda: eq(t.n_atoms, 423)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_pdb_from_url
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 184, in test_pdb_from_url
    eq(t1.n_frames, 1)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.formats.pdb.tests.test_pdb.test_1vii_url_and_gz
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\formats\pdb\tests\test_pdb.py", line 229, in test_1vii_url_and_gz
    eq(t1.n_frames, 1)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.tests.test_netcdf.test_read_after_close
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_netcdf.py", line 48, in <lambda>
    yield lambda: eq(f.n_frames, 101)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.tests.test_trajectory.test_restrict_atoms
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_trajectory.py", line 259, in test_restrict_atoms
    eq(traj.xyz.shape[1], 4)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.tests.test_trajectory.test_restrict_atoms_not_inplace
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_trajectory.py", line 283, in test_restrict_atoms_not_inplace
    eq(sliced.xyz.shape[1], 4)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 
======================================================================
FAIL: mdtraj.tests.test_trajectory.test_stack_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_trajectory.py", line 366, in test_stack_1
    eq(t2.topology._numAtoms, 2*t1.n_atoms)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'int'> <type 'long'>
 
======================================================================
FAIL: mdtraj.tests.test_trajectory.test_stack_2
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python27_64\envs\_test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\tests\test_trajectory.py", line 378, in test_stack_2
    eq(t3.n_atoms, 11)
  File "c:\python27_64\envs\_test\lib\site-packages\mdtraj-1.3.0.dev0-py2.7-win-amd64.egg\mdtraj\testing\testing.py", line 133, in eq
    assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
AssertionError: o1 and o2 not the same type: <type 'long'> <type 'int'>
 ```

--- a/docs/analysis.rst
+++ b/docs/analysis.rst
@@ -57,6 +57,7 @@ Distances
 
     compute_distances
     compute_displacements
+    compute_neighbors
     compute_contacts
     compute_drid
     compute_center_of_mass

--- a/mdtraj/formats/mdcrd.py
+++ b/mdtraj/formats/mdcrd.py
@@ -180,14 +180,14 @@ class MDCRDTrajectoryFile(object):
                                  'supply the number of atoms, "n_atoms"')
             if not os.path.exists(filename):
                 raise IOError("The file '%s' doesn't exist" % filename)
-            self._fh = open(filename, 'r')
+            self._fh = open(filename, 'rb')
             self._is_open = True
             self._fh.readline()  # read comment
             self._line_counter += 1
         elif mode == 'w':
             if os.path.exists(filename) and not force_overwrite:
                 raise IOError("The file '%s' already exists" % filename)
-            self._fh = open(filename, 'w')
+            self._fh = open(filename, 'wb')
             self._is_open = True
         else:
             raise ValueError('mode must be one of "r" or "w". '
@@ -288,7 +288,7 @@ class MDCRDTrajectoryFile(object):
             line = self._fh.readline()
             self._line_counter += 1
 
-            if line == '':
+            if line == b'':
                 raise _EOF()
             try:
                 items = [float(line[j:j+8])
@@ -301,13 +301,11 @@ class MDCRDTrajectoryFile(object):
 
             length = len(items)
 
-            if i + length > len(coords):
-                raise IOError('mdcrd parse error: specified n_atoms (%d) is '
-                              'likely incorrect. Incorrect buffer size '
-                              'encountered. INFO: len(coords)=%d, i=%d, '
-                              'length=%d, items=%s line="%s", coords=%s' % (
-                              self._n_atoms, len(coords), i,  length, items,
-                              line, coords))
+            if i + length > len(coords) or float(line.strip().split()[0]) != items[0]:
+                raise IOError(
+                    'mdcrd parse error: specified n_atoms (%d) is likely incorrect. '
+                    'Incorrect buffer size encountered on line=%d' % (
+                    self._n_atoms, self._line_counter))
 
             coords[i:i+length] = items
             i += length
@@ -319,12 +317,13 @@ class MDCRDTrajectoryFile(object):
                 # peek ahead for box
                 here = self._fh.tell()
                 line = self._fh.readline()
-                peek = [float(elem) for elem in line.split()]
+                peek = [float(elem) for elem in line.strip().split()]
                 if len(peek) == 3:
                     box = peek
                 else:
                     if self._has_box is True:
                         raise IOError('Box information not found in file.')
+                    self._fh.seek(-len(line), 1)
                     self._fh.seek(here)
                 break
 

--- a/mdtraj/formats/mdcrd.py
+++ b/mdtraj/formats/mdcrd.py
@@ -301,7 +301,7 @@ class MDCRDTrajectoryFile(object):
 
             length = len(items)
 
-            if i + length > len(coords) or float(line.strip().split()[0]) != items[0]:
+            if i + length > len(coords):
                 raise IOError(
                     'mdcrd parse error: specified n_atoms (%d) is likely incorrect. '
                     'Incorrect buffer size encountered on line=%d' % (

--- a/mdtraj/formats/mdcrd.py
+++ b/mdtraj/formats/mdcrd.py
@@ -304,7 +304,10 @@ class MDCRDTrajectoryFile(object):
             if i + length > len(coords):
                 raise IOError('mdcrd parse error: specified n_atoms (%d) is '
                               'likely incorrect. Incorrect buffer size '
-                              'encountered. ' % self._n_atoms)
+                              'encountered. INFO: len(coords)=%d, i=%d, '
+                              'length=%d, items=%s line="%s", coords=%s' % (
+                              self._n_atoms, len(coords), i,  length, items,
+                              line, coords))
 
             coords[i:i+length] = items
             i += length

--- a/mdtraj/formats/mdcrd.py
+++ b/mdtraj/formats/mdcrd.py
@@ -31,7 +31,7 @@ import itertools
 import numpy as np
 from mdtraj.utils import ensure_type, cast_indices, in_units_of
 from mdtraj.formats.registry import _FormatRegistry
-from mdtraj.utils.six import string_types
+from mdtraj.utils.six import string_types, PY3
 from mdtraj.utils.six.moves import xrange
 
 __all__ = ['MDCRDTrajectoryFile', 'load_mdcrd']
@@ -357,7 +357,10 @@ class MDCRDTrajectoryFile(object):
         if self._w_has_box is None:
             # this is the first write()
             self._n_atoms = xyz.shape[1]
-            self._fh.write('TITLE : Created by MDTraj with %d atoms\n' % self._n_atoms)
+            comment = 'TITLE : Created by MDTraj with %d atoms\n' % self._n_atoms
+            if PY3:
+                comment = comment.encode('ascii')
+            self._fh.write(comment)
 
             if cell_lengths is None:
                 self._w_has_box = False
@@ -380,16 +383,21 @@ class MDCRDTrajectoryFile(object):
                 out = "%8.3f" % coord
                 if len(out) > 8:
                     raise ValueError('Overflow error')
+                if PY3:
+                    out = out.encode('ascii')
                 self._fh.write(out)
                 if (j+1) % 10 == 0:
-                    self._fh.write("\n")
+                    self._fh.write(b"\n")
                     lfdone = True
 
             if not lfdone:
-                self._fh.write("\n")
+                self._fh.write(b"\n")
 
             if cell_lengths is not None:
-                self._fh.write("%8.3f %8.3f %8.3f\n" % tuple(cell_lengths[i]))
+                line = "%8.3f %8.3f %8.3f\n" % tuple(cell_lengths[i])
+                if PY3:
+                    line = line.encode('ascii')
+                self._fh.write(line)
 
     def seek(self, offset, whence=0):
         """Move to a new file position
@@ -425,7 +433,7 @@ class MDCRDTrajectoryFile(object):
                     self._read()  # advance and throw away these frames
             elif absolute is not None:
                 self._fh.close()
-                self._fh = open(self._filename, 'r')
+                self._fh = open(self._filename, 'rb')
                 self._fh.readline()  # read comment
                 self._frame_index = 0
                 self._line_counter = 1

--- a/mdtraj/geometry/hbond.py
+++ b/mdtraj/geometry/hbond.py
@@ -110,8 +110,8 @@ def wernet_nilsson(traj, exclude_water=True, periodic=True):
     References
     ----------
     .. [1] Wernet, Ph., L.G.M. Pettersson, and A. Nilsson, et al.
-    "The Structure of the First Coordination Shell in Liquid Water." (2004)
-    Science 304, 995-999.
+       "The Structure of the First Coordination Shell in Liquid Water." (2004)
+       Science 304, 995-999.
     """
 
     distance_cutoff = 0.33
@@ -168,6 +168,7 @@ def wernet_nilsson(traj, exclude_water=True, periodic=True):
     # from what's used to compute the angles.
     angle_triplets2 = angle_triplets[:, [1,0,2]]
     return [angle_triplets2[i] for i in mask]
+
 
 def baker_hubbard(traj, freq=0.1, exclude_water=True, periodic=True):
     """Identify hydrogen bonds based on cutoffs for the Donor-H...Acceptor

--- a/mdtraj/testing/testing.py
+++ b/mdtraj/testing/testing.py
@@ -130,6 +130,10 @@ def eq(o1, o2, decimal=6, err_msg=''):
     AssertionError
         If the tests fail
     """
+    if isinstance(o1, integer_types) and isinstance(o2, integer_types) and PY2:
+        eq_(long(o1), long(o2))
+        return
+
     assert (type(o1) is type(o2)), 'o1 and o2 not the same type: %s %s' % (type(o1), type(o2))
 
     if isinstance(o1, dict):
@@ -155,8 +159,6 @@ def eq(o1, o2, decimal=6, err_msg=''):
         assert_dict_equal(o1, o2, decimal=decimal)
     elif isinstance(o1, ast.AST) and isinstance(o2, ast.AST):
         eq_(ast.dump(o1), ast.dump(o2))
-    elif isinstance(o1, integer_types) and isinstance(o2, integer_types) and PY2:
-        eq_(long(o1), long(o2))
 
     # probably these are other specialized types
     # that need a special check?

--- a/mdtraj/testing/testing.py
+++ b/mdtraj/testing/testing.py
@@ -41,7 +41,7 @@ from nose import SkipTest
 from pkg_resources import resource_filename
 
 # py2/3 compatibility
-from mdtraj.utils.six import iteritems
+from mdtraj.utils.six import iteritems, integer_types, PY2
 
 # if the system doesn't have scipy, we'd like
 # this package to still work:
@@ -155,6 +155,8 @@ def eq(o1, o2, decimal=6, err_msg=''):
         assert_dict_equal(o1, o2, decimal=decimal)
     elif isinstance(o1, ast.AST) and isinstance(o2, ast.AST):
         eq_(ast.dump(o1), ast.dump(o2))
+    elif isinstance(o1, integer_types) and isinstance(o2, integer_types) and PY2:
+        eq_(long(o1), long(o2))
 
     # probably these are other specialized types
     # that need a special check?

